### ISSUE TITLE
[Footer] Resolve a11y issues identified by storybook in web components

### ISSUE THREAD

**Issue by jeffchew** [carbon-design-system/carbon-for-ibm-dotcom#7129]
Describe the issue: 

> What did you expect to happen? What happened instead? 

Address all accessibility violations identified in storybook:

![image](https://user-images.githubusercontent.com/1641214/133140677-11216672-4ab6-4575-9077-7c26276ea092.png)

> Component(s) impacted? Please include screenshots, or code and relevant URLs.

Footer in web components
